extensions/amp-access-scroll/0.1/scroll-bar.js

```diff
@@ -0,0 +1,162 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {ScrollComponent} from './scroll-component';
+import {dict} from '../../../src/utils/object';
+
+/**
+ * UI for Scroll users.
+ *
+ * Presents a fixed bar at the bottom of the screen with Scroll content.
+ * @abstract
+ */
+class Bar extends ScrollComponent {
+  /**
+   * @param {!../../../src/service/ampdoc-impl.AmpDoc} doc
+   * @param {!../../amp-access/0.1/amp-access-source.AccessSource} accessSource
+   * @param {string} baseUrl
+   */
+  constructor(doc, accessSource, baseUrl) {
+    super(doc);
+
+    /** @protected */
+    this.accessSource_ = accessSource;
+
+    /** @protected */
+    this.baseUrl_ = baseUrl;
+
+    this.render_();
+  }
+
+  /** @private */
+  render_() {
+    setTimeout(() => {
+      if (!this.frame_) {
+        this.frame_ = this.makeIframe_();
+        this.setWindow_(this.frame_.contentWindow);
+      }
+    }, 0);
+  }
+
+  /**
+   * @return {!HTMLIFrameElement}
+   * @protected
+   * */
+  makeIframe_() {
+    const frame = this.el(
+      'iframe',
+      dict({
+        'scrolling': 'no',
+        'frameborder': '0',
+        'allowtransparency': 'true',
+        'title': 'Scroll',
+        'width': '100%',
+        'height': '100%',
+        'sandbox':
+          'allow-scripts allow-same-origin ' +
+          'allow-top-navigation allow-popups ' +
+          'allow-popups-to-escape-sandbox',
+      })
+    );
+
+    const root = this.el(
+      'div',
+      dict({
+        'class': 'amp-access-scroll-bar',
+      }),
+      [frame]
+    );
+
+    this.mount_(root);
+
+    return /** @type {!HTMLIFrameElement} */ (frame);
+  }
+}
+
+export class ScrollUserBar extends Bar {
+  /**
+   * Add a scrollbar placeholder and then load the scrollbar URL in the iframe.
+   *
+   * @override
+   * */
+  makeIframe_() {
+    const frame = Bar.prototype.makeIframe_.call(this);
+    // Add a placeholder element to display while scrollbar iframe loads.
+    const placeholder = this.el(
+      'div',
+      dict({
+        'class': 'amp-access-scroll-bar amp-access-scroll-placeholder',
```

Comment:
Will this obscure the bottom of the document?

Reply:
Yes, this is the intended behavior for publishers using amp-access-scroll, and is a refactoring of the current behavior in https://github.com/ampproject/amphtml/blob/7ef8d7b13d10a2d08d4eb76283c498ee94d3b618/extensions/amp-access-scroll/0.1/scroll-impl.js#L288-L300 . It creates a fixed toolbar at the bottom of the viewport. Publishers are responsible for ensuring enough space at the bottom of the document to accommodate  this toolbar without obscuring content.